src/litDataset.ts

```diff
@@ -115,10 +115,72 @@ export async function fetchResourceInfo(
 
   const resourceInfo = parseResourceInfo(response);
 
-  return resourceInfo;
+  return { resourceInfo };
 }
 
-function parseResourceInfo(
+/**
+ * @internal
+ *
+ * This (currently internal) function fetches the ACL indicated in the [[WithResourceInfo]]
+ * attached to a resource.
+ * @param resourceInfo The Resource info with the ACL URL
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters).
+ */
+export async function unstable_fetchAcl(
+  resourceInfo: WithResourceInfo,
+  options: Partial<typeof defaultFetchOptions> = defaultFetchOptions
+): Promise<unstable_WithAcl["acl"]> {
+  if (!unstable_hasAccessibleAcl(resourceInfo)) {
+    return {
+      resourceAcl: null,
+      fallbackAcl: null,
+    };
+  }
+  const [resourceAcl, fallbackAcl] = await Promise.all([
+    internal_fetchResourceAcl(resourceInfo, options),
+    internal_fetchFallbackAcl(resourceInfo, options),
+  ]);
+
+  const acl: unstable_WithAcl["acl"] = {
+    fallbackAcl: fallbackAcl,
+    resourceAcl: resourceAcl,
+  };
+
+  return acl;
+}
+
+/**
+ * Experimental: fetch a LitDataset and its associated Access Control List.
+ *
+ * This is an experimental function that fetches both a Resource, the linked ACL Resource (if
+ * available), and the ACL that applies to it if the linked ACL Resource is not available. This can
+ * result in many HTTP requests being executed, in lieu of the Solid spec mandating servers to
+ * provide this info in a single request. Therefore, and because this function is still
+ * experimental, prefer [[fetchLitDataset]] instead.
+ *
+ * If the Resource does not advertise the ACL Resource (because the authenticated user does not have
+ * access to it), the `acl` property in the returned value will be null. `acl.resourceAcl` will be
+ * undefined if the Resource's linked ACL Resource could not be fetched (because it does not exist),
+ * and `acl.fallbackAcl` will be null if the applicable Container's ACL is not accessible to the
+ * authenticated user.
```

Comment:
Corrections: the `acl` object will no longer be null, and `acl.resourceAcl` will not be `undefined` but `null`. (And I guess it doesn't really matter to the developer whether the ACL was advertised or not, just whether they have access to it.)

```suggestion
 * If the Resource's linked ACL Resource could not be fetched (because it does not exist, or because
 * the authenticated user does not have access to it), `acl.resourceAcl` will be `null`. If the
 * applicable Container's ACL is not accessible to the authenticated user, `acl.fallbackAcl` will be
 * `null`.
```